tests/unit/tf/models/test_retrieval.py

```diff
@@ -368,3 +368,37 @@ def last_interaction_as_target(inputs, targets):
     losses = model.fit(dataloader, epochs=1)
 
     assert losses is not None
+
+
+@pytest.mark.parametrize("run_eagerly", [True, False])
+def test_to_recommender_v2(ecommerce_data: Dataset, run_eagerly):
+    def _item_id_as_target(inputs, targets):
+        items = inputs["item_id"]
+        return inputs, items
+
+    dataloader = BatchedDataset(ecommerce_data, batch_size=50)
+    dataloader = dataloader.map(_item_id_as_target)
+
+    model = mm.TwoTowerModelV2(ecommerce_data.schema, query_tower=mm.MLPBlock([64, 128]))
+    model.compile(run_eagerly=run_eagerly, optimizer="adam")
+    _ = model.fit(ecommerce_data, batch_size=50, epochs=5, steps_per_epoch=1)
+
+    recommender = model.to_item_recommender(dataset=ecommerce_data, batch_size=50, k=10)
+    recommender.compile(run_eagerly=run_eagerly, optimizer="adam")
```

Comment:
Do we need to set the optimizer of item_recommender if it is just to be used for evaluation and prediction?